tests/munging/test-delete.py

```diff
@@ -316,14 +316,27 @@ def test_del_rows_array():
     assert d0.to_list() == [[1, 2, 3, 4, 5, 6, 9]]
 
 
-@pytest.mark.skip("Issue #805")
 def test_del_rows_array_unordered():
     d0 = dt.Frame(range(10))
     del d0[[3, 1, 5, 2, 2, 0, -1], :]
     frame_integrity_check(d0)
     assert d0.to_list() == [[4, 6, 7, 8]]
 
 
+def test_del_rows_array_unordered_index_duplicated():
+    d0 = dt.Frame(range(10))
+    del d0[[-1, 3, 0, 0, 3, -1, 4, 1, 1], :]
+    frame_integrity_check(d0)
+    assert d0.to_list() == [[2, 5, 6, 7, 8]]
+
+
+def test_del_rows_array_unordered_larger_than_frame():
+    d0 = dt.Frame(range(10))
+    del d0[[5, 6, 5, 8, 5, -1, 3, 0, 0, 3, -1, 4, 1, 1], :]
+    frame_integrity_check(d0)
+    assert d0.to_list() == [[2, 7]]
+
+
```

Comment:
I would also suggest to add the following tests: 
- delete a one-element list of rows (e.g. `[3]`);
- delete an empty list of rows (`[]`);
- delete a list of rows, possibly with repetitions, such that all rows are removed from the frame (e.g. `[3, 2, 0, 2, 0, 1, 4, 4, 0]` with a 5-rows frame);
- delete a long list of identical rows (e.g. `[3]*1000`), see my comment above.

Reply:
I agree